### DISCUSSION TITLE
Issue mocking family async notifier

### DISCUSSION THREAD

**Discussion by imtoori** [rrousselGit/riverpod#2665] [Q&A]
I'm having an issue while mocking an AsyncNotifier

My controller is something like this 
```dart
@riverpod
class AlertGroupsController extends _$AlertGroupsController {
  @override
  Future<AlertGroupsUiState> build(AGListFilter? filter) async {
    final alertGroupRepository = await ref.watch(AlertGroupRepository.provider);
    final alertGroups = await alertGroupRepository.getLocalAlertGroups(
      filter: filter,
    );

    return AlertGroupsUiState(alertGroups: alertGroups);
  }
...
```

This is my mock
```dart

class MockAlertGroupsController
    extends AutoDisposeFamilyAsyncNotifier<AlertGroupsUiState, AGListFilter?>
    with Mock
    implements AlertGroupsController {
  @override
  Future<AlertGroupsUiState> build(AGListFilter? filter) async {
    return const AlertGroupsUiState(alertGroups: []);
  }

  @override
  Future<void> getMoreAlertGroups() async {}
}

```

During tests I get this error

```
Error: type 'MockAlertGroupsController' is not a subtype of type 'AutoDisposeAsyncNotifier<AlertGroupsUiState>' in type cast
```

Following the stacktrace this happens right when I call ref.read/watch on the generated provider.

Any ideas on how to fix this? I managed to solve

**Comment by rrousselGit** [comment]
You shouldn't extend AutoDisposeFamilyAsyncNotifier but _$AlertGroupsController

**Reply by imtoori** [reply]
but isn't _$AlertGroupsController private?

**Reply by rrousselGit** [reply]
Define your mock in the same file

**Reply by imtoori** [reply]
I'll give it a try, but you agree it's a bit weird defining the mock in the same file as the controller?

**Reply by rrousselGit** [reply]
Not really. That's kind of needed in many cases in Dart 3 anyway due to keywords like `sealed`/`final`/...

**Reply by imtoori** [reply]
makes sense and it's not a big deal, but I'm still getting the same error 🤔

**Reply by imtoori** [reply]
can you take a look at this repository? https://github.com/imtoori/test_riverpod

it's a demo project for showing my error. If you run `flutter test` you'll get what I experience. 

Oh and thanks a lot for the support!!!
